File: go/problems/day1_test.go
package problems

import "testing"

var day1 Solvable[int] = Day1{}

func TestDay1_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day1_1")
	result := day1.Solve(input)

	expected := 142
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay1_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day1_2")
	result := day1.Solve(input)

	expected := 55017
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day1.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import { solve } from "./day1.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 1", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day1_1");
    const result = solve(input);
    assert.equal(result, 142);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day1_2");
    const result = solve(input);
    assert.equal(result, 55017);
  });
});
